src/device/device_default.cpp

```diff
@@ -194,16 +195,21 @@ namespace hw {
             return address;
         }
 
-        crypto::secret_key  device_default::get_subaddress_secret_key(const crypto::secret_key &a, const cryptonote::subaddress_index &index) {
+        void device_default::get_subaddress_secret_key(const crypto::secret_key &a, const cryptonote::subaddress_index &index, crypto::secret_key &skey) {
```

Comment:
What's the purpose of replacing a return value for a reference-out parameter? It makes the code less readable.